extensions.html

```diff
@@ -1,920 +1,1152 @@
+          new {{Gamepad}} object is created.
+        </dd>
+        <dt>
+          <dfn>surfaceId</dfn>
+        </dt>
+        <dd>
+          Unique id of the surface that generated the touch event.
+        </dd>
+        <dt>
+          <dfn>position</dfn>
+        </dt>
+        <dd>
+          x, y coordinates of the touch event. Range of each coordinate is
+          normalized to [-1.0, 1.0]. Along the x-axis, -1.0 references the
+          leftmost coordinate and 1.0 references the rightmost coordinate.
+          Along the y-axis, -1.0 references the topmost coordinate and 1.0
+          references the bottommost coordinate. MUST be a two-element array.
+        </dd>
+        <dt>
+          <dfn>surfaceDimensions</dfn>
+        </dt>
+        <dd>
+          Width and height of the touch surface in integer units. When not
+          <code>null</code>, MUST be a two-element array.
+        </dd>
+      </dl>
+    </section>
+    <section>
+      <h2>
+        Partial <dfn>Gamepad</dfn> Interface
+      </h2>
+      <p>
+        This partial interface supplements the Gamepad interface described in
+        the main <a href="https://w3c.github.io/gamepad/">Gamepad spec</a>.
+      </p>
+      <pre class="idl">
+        partial interface Gamepad {
+          readonly attribute GamepadHand hand;
+          readonly attribute FrozenArray&lt;GamepadHapticActuator&gt; hapticActuators;
+          readonly attribute GamepadPose? pose;
+          readonly attribute FrozenArray&lt;GamepadTouch&gt;? touchEvents;
+          [SameObject] readonly attribute GamepadHapticActuator vibrationActuator;
+        };
+      </pre>
+      <p>
+        Instances of {{Gamepad}} are created with the internal slots described
+        in the following table:
+      </p>
+      <table class="simple">
+        <tr>
+          <th>
+            Internal slot
+          </th>
+          <th>
+            Initial value
+          </th>
+          <th>
+            Description (non-normative)
+          </th>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for="Gamepad">[[\hand]]</dfn>
+          </td>
+          <td>
+            `undefined`
+          </td>
+          <td>
+            Indicates the hand the controller is held in.
+          </td>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for="Gamepad">[[\hapticActuators]]</dfn>
+          </td>
+          <td>
+            `undefined`
+          </td>
+          <td>
+            List of all the haptic actuators in the gamepad.
+          </td>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for="Gamepad">[[\pose]]</dfn>
+          </td>
+          <td>
+            `null`
+          </td>
+          <td>
+            The current pose of the gamepad.
+          </td>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for="Gamepad">[[\touchEvents]]</dfn>
+          </td>
+          <td>
+            <code>null</code>

```

Comment:
```suggestion
            `null`
```

Reply:
I wonder if this should always just be an empty list?

Reply:
I think because not all Gamepads support touch pads, it is better this way to distinguish between a Gamepad that will never have touchIds (null) and a Gamepad that just has no new input (empty list).